department-of-veterans-affairs/abd-vro
# EP Merge: Add claim contention metrics

## What was the problem?

EP Merge app does not publish metrics on the number or type of contentions on a pending EP or EP400.  The only metric related to this is the number of contentions that were merged.

Associated tickets or Slack threads:
- #3065

## How does this fix it?

This adds metrics that capture the following: 
* `job.merged_contentions` - previously captured as `job.new_contentions`
* `job.merged_new_contentions` - # of new contentions on the merged claim
* `job.merged_cfi_contentions` - # of cfi contentions on the merged claim
* `pending_ep.new_contentions` - # of new contentions on the pending claim
* `pending_ep.cfi_contentions` - # of cfi contentions on the pending claim
* `ep400.new_contentions` - # of new contentions on the ep400 claim
* `ep400.cfi_contentions` - # of cfi contentions on the ep400 claim

## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES=“all” ./gradlew dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`
